src/game/game_text.cpp

```diff
@@ -273,6 +274,31 @@ GameStrings *LoadTranslations()
 	}
 }
 
+static StringParam::ParamType GetParamType(const CmdStruct *cs)
+{
+	if (cs->value == SCC_RAW_STRING_POINTER) return StringParam::RAW_STRING;
+	if (cs->value == SCC_STRING || cs != TranslateCmdForCompare(cs)) return StringParam::STRING;
+	return StringParam::OTHER;
+}
+
+static void ExtractStringParams(const StringData &data, StringParamsList &params)
+{
+	for (size_t i = 0; i < data.max_strings; i++) {
+		const LangString *ls = data.strings[i];
+
+		if (ls != nullptr) {
+			params.emplace_back();
```

Comment:
It looks a bit weird to add basically nothing to the `params` to me. I know it's allocating a vector, however the `params.back()` looks a bit weird as well.
Maybe just `auto param = params.emplace_back();` and `param.emplace_back(...` later on?